airflow/jobs.py

```diff
@@ -547,41 +785,105 @@ def process_dag(self, dag, queue):
 
                 if ti.is_runnable(flag_upstream_failed=True):
                     self.logger.debug('Queuing task: {}'.format(ti))
-                    queue.put((ti.key, pickle_id))
+                    queue.put(ti.key)
 
         session.close()
 
     @provide_session
-    def prioritize_queued(self, session, executor, dagbag):
-        # Prioritizing queued task instances
+    def _change_state_for_tis_without_dagrun(self,
+                                             simple_dag_bag,
+                                             old_states,
+                                             new_state,
+                                             session=None):
+        """
+        For all DAG IDs in the SimpleDagBag, look for task instances in the
+        old_states and set them to new_state if the corresponding DagRun
+        exists but is not in the running state.
+
+        :param old_states: examine TaskInstances in this state
+        :type old_state: list[State]
+        :param new_state: set TaskInstances to this state
+        :type new_state: State
+        :param simple_dag_bag: TaskInstances associated with DAGs in the
+        simple_dag_bag and in the UP_FOR_RETRY state will be examined.
+        :type simple_dag_bag: SimpleDagBag
+        """
 
-        pools = {p.pool: p for p in session.query(models.Pool).all()}
+        task_instances_to_change = (
+            session
+            .query(models.TaskInstance)
+            .filter(models.TaskInstance.dag_id.in_(simple_dag_bag.dag_ids))
+            .filter(models.TaskInstance.state.in_(old_states))
+            .all()
+        )
+        """:type: list[TaskInstance]"""
+
+        for task_instance in task_instances_to_change:
+            dag_run = DagRun.get_run(session,
```

Comment:
get_run naming implies you will always get a dag_run. Why are you not using "DagRun.find(..)"? If you are always expecting a return make get_run (or load_run) use ".one()" instead of ".first()"